src/ingress/models.py

```diff
@@ -0,0 +1,29 @@
+from django.db import models
+
+
+class IngressQueue(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+    endpoint = models.ForeignKey('Endpoint', on_delete=models.CASCADE)
+    raw_data = models.TextField()  # Store raw received data to be parsed by a separate parser
+    parse_started = models.DateTimeField(null=True)
+    parse_succeeded = models.DateTimeField(null=True)
+    parse_failed = models.DateTimeField(null=True)
+    parse_fail_info = models.TextField(null=True)  # To store stack traces or other info about the fail
+
+
+class FailedIngressQueue(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+    endpoint = models.ForeignKey('Endpoint', on_delete=models.CASCADE)
+    raw_data = models.TextField()  # Store raw received data to be parsed by a separate parser
+    parse_started = models.DateTimeField(null=True)
+    parse_succeeded = models.DateTimeField(null=True)
+    parse_failed = models.DateTimeField(null=True)
+    parse_fail_info = models.TextField(null=True)  # To store stack traces or other info about the fail
+
+
+class Endpoint(models.Model):
```

Comment:
Hoe ga je deze toevoegen? Als er een interface voor is, dan is het heel handig, maar als het enkel in de db bestaat op prod, dan kunnen we er niet bij. 

Dan zouden we ook uit de voeten kunnen met een array in python code. Een alternatief is dat je de endpoint open laat, en alles op `/ingress` opslaat inclusief endpoint. Je kan dan alerting instellen wanneer er niet-verwachtte endpoints aangeroepen worden. Dan slaan we dus wel de data op, maar weten we gelijk dat we er iets mee moeten doen (parser voor schrijven, of terugkoppeling doen naar leverancier).

Reply:
Een alternatief is dat je helemaal niet opslaat welke endpoints bestaan, maar dat je applicaties tijdens de app loading zichzelf laat registreren en dan kenbaar laat maken welke endpoints ze verantwoordelijk voor zijn. Komt er dan een request binnen waar niemand verantwoordelijk voor is, dan weten we dat we iets moeten doen.

Reply:
Nadeel van dit alles is dan wel weer dat je niet de endpoints kan afdwingen. Gezien de geschiedenis van dit project is dat misschien wel fijn om te kunnen. Anders zullen we snel wildgroei krijgen en inconsistentie. Moeten we even verder bespreken

Reply:
Er zijn management commands voor het toevoegen en verwijderen van endpoints (`add_endpoint` en `remove_endpoint`). Dit zorg ervoor dat het toevoegen ook mogelijk is op prod. Wellicht moeten we ook nog een list command ervoor hebben om er inzicht in te kunnen krijgen.

Als een endpoint niet bestaat dan krijgt de verzender gewoon een 404. Dat lijkt me ook de correcte manier van doen. Alles opslaan wat maar naar `/ingress/whatever` wordt gestuurd lijkt me alleen maar troep creëren, al was het maar door geautomatiseerde hack scripts die bv phpmyadmin aanroepen.

Maar laten we deze idd nog bespreken.